### PR TITLE
t.rast.line: add option to decide on x-value

### PR DESCRIPTION
**Pull request by ecodiv**
When raster values represent time ranges, select start, mid or end dates as x-value.

### CODE REVIEW

File: src/temporal/t.rast.line/t.rast.line.html [hunk 1]
```diff
@@ -52,8 +52,8 @@ <h2>DESCRIPTION</h2>
 <p>
 The <b>where</b> options allows allows performing different selections 
 of maps registered in the space-time datasets. For example, with 
-<i>start_time < '2020-01-01'</i> the time series is limited to all maps 
-with a start time before the given date. For more details, see <a 
+<i>start_time &lt; '2020-01-01'</i> the time series is limited to all 
+maps with a start time before the given date. For more details, see <a 
 href="https://grass.osgeo.org/grass83/manuals/temporalintro.html#modules-to-process-space-time-raster-datasets">this 
 page</a> for more details.
```
Comment (veroandreo) [inline src/temporal/t.rast.line/t.rast.line.html:58]:
```suggestion
page</a>.
```

**Review by veroandreo — COMMENTED** [review]
I tested with following command:

```
t.rast.line -l -g input=tempmean zones=comparison_counties error=sd nprocs=10     where="start_time > '2004-01-01' AND start_time < '2005-01-01'" x_value=xx
```
- x_value=start
image

- x_value=mid
image

- x_value=end
image

First and last labels get no month, is that expected? Also the label March in the one for x_value=end doesn't seem right to me. Is it switched with start? Am I reading it right?

**Comment by ecodiv** [comment]
Hi Vero, thanks for testing. Can't try out myself right now. Can you try to set the date format to month or even weeks and see how that looks like?

**Comment by veroandreo** [comment]
> Hi Vero, thanks for testing. Can't try out myself right now. Can you try to set the date format to month or even weeks and see how that looks like?

Labels look better, but still, they are placed at the begining of the interval instead the end as indicated in the command:
```
t.rast.line -l -g input=tempmean zones=comparison_counties error=sd nprocs=10 where="start_time > '2004-01-01' AND start_time < '2005-01-01'" x_value=end date_format="%Y-%m"
```
image

and at the end when they should be at the beginning:

image

Also, note the selection is for larger than so Jan should not appear there. But now I remember there was a bug in sql I believe, so to get the right selection with `>` date and time need to be explicit, i.e., 

```
GRASS nc_basic_spm_grass7/climate_2000_2012:~ > t.rast.list input=tempmean where="start_time > '2004-01-01' AND start_time < '2005-01-01'"
name|mapset|start_time|end_time
2004_01_tempmean|climate_2000_2012|2004-01-01 00:00:00|2004-02-01 00:00:00
2004_02_tempmean|climate_2000_2012|2004-02-01 00:00:00|2004-03-01 00:00:00
2004_03_tempmean|climate_2000_2012|2004-03-01 00:00:00|2004-04-01 00:00:00
2004_04_tempmean|climate_2000_2012|2004-04-01 00:00:00|2004-05-01 00:00:00
2004_05_tempmean|climate_2000_2012|2004-05-01 00:00:00|2004-06-01 00:00:00
2004_06_tempmean|climate_2000_2012|2004-06-01 00:00:00|2004-07-01 00:00:00
2004_07_tempmean|climate_2000_2012|2004-07-01 00:00:00|2004-08-01 00:00:00
2004_08_tempmean|climate_2000_2012|2004-08-01 00:00:00|2004-09-01 00:00:00
2004_09_tempmean|climate_2000_2012|2004-09-01 00:00:00|2004-10-01 00:00:00
2004_10_tempmean|climate_2000_2012|2004-10-01 00:00:00|2004-11-01 00:00:00
2004_11_tempmean|climate_2000_2012|2004-11-01 00:00:00|2004-12-01 00:00:00
2004_12_tempmean|climate_2000_2012|2004-12-01 00:00:00|2005-01-01 00:00:00
GRASS nc_basic_spm_grass7/climate_2000_2012:~ > t.rast.list input=tempmean where="start_time > '2004-01-01 00:00:00' AND start_time < '2005-01-01'"
name|mapset|start_time|end_time
2004_02_tempmean|climate_2000_2012|2004-02-01 00:00:00|2004-03-01 00:00:00
2004_03_tempmean|climate_2000_2012|2004-03-01 00:00:00|2004-04-01 00:00:00
2004_04_tempmean|climate_2000_2012|2004-04-01 00:00:00|2004-05-01 00:00:00
2004_05_tempmean|climate_2000_2012|2004-05-01 00:00:00|2004-06-01 00:00:00
2004_06_tempmean|climate_2000_2012|2004-06-01 00:00:00|2004-07-01 00:00:00
2004_07_tempmean|climate_2000_2012|2004-07-01 00:00:00|2004-08-01 00:00:00
2004_08_tempmean|climate_2000_2012|2004-08-01 00:00:00|2004-09-01 00:00:00
2004_09_tempmean|climate_2000_2012|2004-09-01 00:00:00|2004-10-01 00:00:00
2004_10_tempmean|climate_2000_2012|2004-10-01 00:00:00|2004-11-01 00:00:00
2004_11_tempmean|climate_2000_2012|2004-11-01 00:00:00|2004-12-01 00:00:00
2004_12_tempmean|climate_2000_2012|2004-12-01 00:00:00|2005-01-01 00:00:00
```

Still, label position seems inverted between start and end options.

**Comment by ecodiv** [comment]
Thanks. I'll have a look when back

**Comment by ecodiv** [comment]
@veroandreo Just tested it, and it seems to work as intended. By default, the AutoDateLocator of matplotlib (https://matplotlib.org/stable/api/dates_api.html#matplotlib.dates.AutoDateLocator) is used. Can you try to set the date_interval to day, and check again (the week option does not work, fixing that now)

**Comment by veroandreo** [comment]
> @veroandreo Just tested it, and it seems to work as intended. By default, the AutoDateLocator of matplotlib (https://matplotlib.org/stable/api/dates_api.html#matplotlib.dates.AutoDateLocator) is used. Can you try to set the date_interval to day, and check again (the week option does not work, fixing that now)

Data is monthly so setting the interval to day, didn't really help. But I tested again, and indeed it works fine. Seems all the confusion arose because of the selection bug when using greater than (`>`) and only a date; *date and time* is needed to get the right output, i.e.: 

```
t.rast.line -l -g input=tempmean zones=comparison_counties error=sd nprocs=10 where="start_time > '2004-01-01 00:00:00' AND start_time < '2005-01-01'" x_value=start date_format="%Y-%m" date_interval=month
t.rast.line -l -g input=tempmean zones=comparison_counties error=sd nprocs=10 where="start_time > '2004-01-01 00:00:00' AND start_time < '2005-01-01'" x_value=end date_format="%Y-%m" date_interval=month
```

**Comment by ecodiv** [comment]
> > @veroandreo Just tested it, and it seems to work as intended. By default, the AutoDateLocator of matplotlib (https://matplotlib.org/stable/api/dates_api.html#matplotlib.dates.AutoDateLocator) is used. Can you try to set the date_interval to day, and check again (the week option does not work, fixing that now)
> 
> Data is monthly so setting the interval to day, didn't really help. But I tested again, and indeed it works fine. Seems all the confusion arose because of the selection bug when using greater than (`>`) and only a date; _date and time_ is needed to get the right output, i.e.:
> 
> ```
> t.rast.line -l -g input=tempmean zones=comparison_counties error=sd nprocs=10 where="start_time > '2004-01-01 00:00:00' AND start_time < '2005-01-01'" x_value=start date_format="%Y-%m" date_interval=month
> t.rast.line -l -g input=tempmean zones=comparison_counties error=sd nprocs=10 where="start_time > '2004-01-01 00:00:00' AND start_time < '2005-01-01'" x_value=end date_format="%Y-%m" date_interval=month
> ```

Thanks for checking out, and I'll add these observations to the help file.

**Pull request merged**